wpilibc/src/main/native/cpp/Filters/Filter.cpp

```diff
@@ -9,14 +9,16 @@
 
 using namespace frc;
 
-Filter::Filter(std::shared_ptr<PIDSource> source) { m_source = source; }
+Filter::Filter(PIDSource& source) : m_source(source) {}
+
+Filter::Filter(std::shared_ptr<PIDSource> source) : m_source(*source) {}
```

Comment:
This seems really dangerous.  What happens when someone constructs a filter with `Filter(std::make_shared)`?  It will get immediately freed.